### PR TITLE
Perform one KDF benchmark at a time

### PR DESCRIPTION
**Pull request by droidmonkey**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
* Prevent using double memory when benchmarking Argon2 performance.
* Improve benchmark results by not overusing CPU and memory simultaneously. Parallel benchmarks were causing artificially strained calculations resulting in a lower than desired number of rounds.

Thanks to @Piraty for pointing this out!!

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested manually with 1GB Argon2 memory setting; memory of app only increased by 1GB during benchmarking.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

**Comment by droidmonkey** [comment]
Yah we can try that as well, good idea. Just double the rounds returned.

**Comment by droidmonkey** [comment]
Implemented

**Comment by phoerious** [comment]
Or add them and don't divide by two. 😉

File: src/crypto/kdf/Kdf.cpp [hunk 1]
```diff
@@ -71,16 +71,19 @@ void Kdf::randomizeSeed()
 
 int Kdf::benchmark(int msec) const
 {
-    BenchmarkThread thread1(msec, this);
-    BenchmarkThread thread2(msec, this);
+    // Run the benchmark twice using half the time for each run
+    BenchmarkThread thread(msec/2, this);
```
Comment (phoerious) [inline src/crypto/kdf/Kdf.cpp:75]: Could you add spaces here, pretty plz?

Comment (droidmonkey) [reply src/crypto/kdf/Kdf.cpp]: Done!

**Comment by Piraty** [comment]
lgtm

**Pull request merged**